verification/cbmc/sources/make_common_data_structures.c

```diff
@@ -126,21 +127,55 @@ bool aws_cryptosdk_edk_list_elements_are_bounded(const struct aws_array_list *co
 }
 
 void ensure_cryptosdk_edk_list_has_allocated_list(struct aws_array_list *list) {
-    if (list->current_size == 0) {
-        __CPROVER_assume(list->data == NULL);
-        list->alloc = can_fail_allocator();
-    } else {
-        size_t max_length = list->current_size / sizeof(struct aws_cryptosdk_edk);
-        list->data        = bounded_malloc(sizeof(struct aws_cryptosdk_edk) * max_length);
-        list->alloc       = nondet_bool() ? NULL : can_fail_allocator();
+    if (list != NULL) {
+        if (list->current_size == 0) {
+            __CPROVER_assume(list->data == NULL);
+            list->alloc = can_fail_allocator();
+        } else {
+            size_t max_length = list->current_size / sizeof(struct aws_cryptosdk_edk);
+            list->data        = bounded_malloc(sizeof(struct aws_cryptosdk_edk) * max_length);
+            list->alloc       = nondet_bool() ? NULL : can_fail_allocator();
+        }
     }
 }
 
 void ensure_cryptosdk_edk_list_has_allocated_list_elements(struct aws_array_list *list) {
-    for (size_t i = 0; i < list->length; ++i) {
-        struct aws_cryptosdk_edk *data = (struct aws_cryptosdk_edk *)list->data;
-        ensure_cryptosdk_edk_has_allocated_members(&(data[i]));
+    if (aws_cryptosdk_edk_list_is_valid(list)) {
+        for (size_t i = 0; i < list->length; ++i) {
+            struct aws_cryptosdk_edk *data = (struct aws_cryptosdk_edk *)list->data;
+            ensure_cryptosdk_edk_has_allocated_members(&(data[i]));
+        }
+    }
+}
+
+struct aws_cryptosdk_hdr *ensure_nondet_hdr_has_allocated_members(const size_t max_table_size) {
+    struct aws_cryptosdk_hdr *hdr = malloc(sizeof(*hdr));
+    if (hdr != NULL) {
+        hdr->alloc     = nondet_bool() ? NULL : can_fail_allocator();
+        hdr->frame_len = malloc(sizeof(hdr->frame_len));
```

Comment:
if `frame_len` is not a pointer, you don't need that. https://github.com/aws/aws-encryption-sdk-c/blob/fc974765d986662589e8c9f1a07eae6e9bf83665/include/aws/cryptosdk/private/header.h#L29

Reply:
Updated!